vllm/distributed/parallel_state.py

```diff
@@ -431,20 +431,45 @@ def gather(self,
         if dim < 0:
             # Convert negative dim to positive.
             dim += input_.dim()
-        # Allocate output tensor.
-        if self.rank_in_group == dst:
-            gather_list = [torch.empty_like(input_) for _ in range(world_size)]
-        else:
-            gather_list = None
-        # Gather.
-        torch.distributed.gather(input_,
-                                 gather_list,
-                                 dst=self.ranks[dst],
-                                 group=self.device_group)
-        if self.rank_in_group == dst:
-            output_tensor = torch.cat(gather_list, dim=dim)
```

Comment:
you don't need to change so many lines for the rest platforms. just add:

```python
if current_platform.is_xpu():
    output = self.all_gather(xxx)
    if self.rank == dst:
        return output
```

please minimize the code change to common files.

Reply:
Updated. Thanks!